pkg/compact/compact.go

```diff
@@ -992,8 +1050,13 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 
 	var toCompact []*metadata.Meta
 	if err := tracing.DoInSpanWithErr(ctx, "compaction_planning", func(ctx context.Context) (e error) {
-		toCompact, e = planner.Plan(ctx, cg.metasByMinTime)
-		return e
+		if cg.partitionInfo.PartitionCount > 0 {
```

Comment:
Can partitionInfo be `nil`?

Reply:
Yes, it can be `nil`. I forget to add check here.